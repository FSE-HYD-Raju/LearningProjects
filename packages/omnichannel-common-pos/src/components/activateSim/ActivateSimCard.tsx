import * as React from "react";
import {
	ActivateSimConfigurationType,
	ActivateSimOrderData,
	Order,
	OrderItem,
	User,
	SimIccVerification,
	SimIccVerificationAttributes
} from "../../redux/types";
import ActivateSimBanner from "./ActivateSimBanner";
import OrderUtil from "../../utils/order/OrderUtil";

export interface ActivateSimCardStateProps {
	activateSimOrderData?: ActivateSimOrderData;
	user?: User;
	configuration: ActivateSimConfigurationType;
	verificationResponse?: SimIccVerification;
	verificationError: boolean;
}

export interface ActivateSimCardActionProps {
	actions: {
		getInProgressOrders: (personId: string) => void;
		simIccVerification: (payload: SimIccVerificationAttributes) => void;
		onClose: () => void;
	};
}

export type ActivateSimCardProps = ActivateSimCardStateProps & ActivateSimCardActionProps;

export type SimCardActivationInfo = {
	iccid: string;
	orderReference: string;
	tocTocUrl: string;
	iccidVerification: boolean;
};

class ActivateSimCard extends React.PureComponent<ActivateSimCardProps> {
	constructor(props: ActivateSimCardProps) {
		super(props);
	}

	componentWillMount() {
		const { user, actions } = this.props;
		if (user && user.id) {
			actions.getInProgressOrders(user.id);
		}
	}

	getSimCardActivationInfo = (): SimCardActivationInfo[] => {
		const { activateSimOrderData, configuration } = this.props;
		const orders = activateSimOrderData && activateSimOrderData.orders;

		const result: SimCardActivationInfo[] = [];

		if (this.isConfigurationValid() && orders && orders.length) {
			orders.forEach((order: Order) => {
				const orderDetails = order.attributes && order.attributes.orderDetails;
				const orderReference = order.attributes && order.attributes.referenceNumber;

				const iccid =
					orderDetails &&
					orderDetails.msisdnIccid &&
					orderDetails.msisdnIccid.length &&
					orderDetails.msisdnIccid[0].iccid;

				const token = orderDetails && orderDetails.token;

				if (iccid && orderReference) {
					const allOrderItems = OrderUtil.getAllOrderItems(order);

					const activationItem = allOrderItems.find(item =>
						this.isItemContainsOneOfPOs(item, configuration.activatePOIds as string[])
					);

					const deliveryItem = allOrderItems.find(item =>
						this.isItemContainsOneOfPOs(item, configuration.deliveryPOIds as string[])
					);

					if (activationItem && deliveryItem) {
						const vcn = configuration.verificationCharacteristicName as string;
						const iccidVv = configuration.iccidVerificationValue as string;
						const videoVv = configuration.videoVerificationValue as string;

						const iccidVerification = this.isItemContainsCharacteristicWithValue(
							deliveryItem,
							vcn,
							iccidVv
						);
						const videoVerification = this.isItemContainsCharacteristicWithValue(
							deliveryItem,
							vcn,
							videoVv
						);

						if (iccidVerification || videoVerification) {
							result.push({
								iccid,
								orderReference,
								iccidVerification,
								tocTocUrl: `${token}`
							});
						}
					}
				}
			});
		}

		return result;
	};

	isConfigurationValid = () => {
		const { configuration } = this.props;

		return (
			configuration &&
			configuration.activatePOIds &&
			configuration.activatePOIds.length &&
			configuration.deliveryPOIds &&
			configuration.deliveryPOIds.length &&
			configuration.verificationCharacteristicName &&
			configuration.iccidVerificationValue &&
			configuration.videoVerificationValue
		);
	};

	isItemContainsOneOfPOs = (orderItem: OrderItem, pos: string[]): boolean => {
		const po = (orderItem.attributes && orderItem.attributes.productOffering) || orderItem.productOffering;

		return Boolean(po && pos.includes(po.id));
	};

	isItemContainsCharacteristicWithValue = (
		orderItem: OrderItem,
		characteristicName: string,
		characteristicValue: string
	): boolean => {
		const characteristics =
			(orderItem.attributes && orderItem.attributes.inputtedCharacteristics) || orderItem.inputtedCharacteristics;

		return Boolean(characteristics && characteristics[characteristicName] === characteristicValue);
	};

	render() {
		const { user } = this.props;

		const simCardActivationInfo = this.getSimCardActivationInfo();

		if (user && user.attributes && simCardActivationInfo.length) {
			const name = user.attributes.firstName;

			return (
				<ActivateSimBanner
					name={name}
					simCardActivationInfo={simCardActivationInfo}
					verificationResponse={this.props.verificationResponse}
					verificationError={this.props.verificationError}
					simIccVerification={this.props.actions.simIccVerification}
					onClose={this.props.actions.onClose}
				/>
			);
		} else {
			return null;
		}
	}
}

export default ActivateSimCard;
