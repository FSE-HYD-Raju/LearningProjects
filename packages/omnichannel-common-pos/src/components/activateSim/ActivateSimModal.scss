.ActivateSimModal {
	&-container {
		display: flex;
		flex-wrap: wrap;
		width: 100%;
	}
	&-verification-block {
		width: 100%;
		display: flex;
		align-items: center;
		margin: 0.5rem 0 1rem;
	}
	&-label {
		min-width: 10.25rem;
		color: $gray-600;
	}
	&-sim-block {
		width: 100%;
		margin-left: 10.25rem;
	}
	&-inputNormal {
		max-width: 7.5rem;
		height: 2.5rem;
		padding: 0 0.3rem;
		outline: none;
		border: 0.0625rem solid $weasel-border-color-solid;
		&:active {
			outline: none;
		}
	}
	&-error {
		border: 0.0625rem solid $danger;
	}

	&-error-alert {
		position: absolute;
		bottom: 0.625rem;
		display: flex;
		align-items: center;
		flex-wrap: nowrap;
		color: red;

		i {
			margin-right: 0.3125rem;
		}
	}
}

button#activate-sim-cancel-button {
	padding: 0.565rem;
}
