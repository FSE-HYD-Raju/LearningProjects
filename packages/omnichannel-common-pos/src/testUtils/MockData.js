import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
