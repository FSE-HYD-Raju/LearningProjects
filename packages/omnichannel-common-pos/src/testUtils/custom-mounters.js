import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
