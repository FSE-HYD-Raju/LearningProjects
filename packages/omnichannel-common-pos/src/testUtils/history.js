import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
