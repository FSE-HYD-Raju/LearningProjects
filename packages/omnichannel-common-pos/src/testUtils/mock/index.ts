import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
