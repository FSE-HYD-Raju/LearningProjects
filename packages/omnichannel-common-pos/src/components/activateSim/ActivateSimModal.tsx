import * as R from "react";
import FormattedMessage from "../../channelUtils/FormattedMessage";
import activateSimMessages from "./activateSim.messages";
import classnames from "classnames";
import OcModal from "../ocComponents/OcModal";
import SimExampleView from "./simExampleView/SimExampleView";
import { SimCardActivationInfo } from "./ActivateSimCard";
import cssns from "../../utils/cssnsConfig";
import { ContextType, contextTypesValidationMap } from "../../types";
import { SimIccVerification, SimIccVerificationAttributes } from "../../redux";
import { CommonCustomizationPoints, withCustomization } from "../../customization";

const React = cssns("ActivateSimModal").React as typeof R;

interface ActivateSimModalProps {
	simCardActivationInfo: SimCardActivationInfo[];
	show: boolean;
	closeModal: () => void;
	verificationResponse?: SimIccVerification;
	verificationError: boolean;
	simIccVerification: (payload: SimIccVerificationAttributes) => void;
}

interface ActivateSimModalState {
	value: string;
	error: boolean;
	buttonDisabled: boolean;
	showIccidAlert: boolean;
	showVerificationAlert: boolean;
	verifying: boolean;
}

const ERROR_RE = /^\d{0,5}$/;
const DISABLED_RE = /^\d{5}$/;

const initialState: ActivateSimModalState = {
	value: "",
	error: false,
	buttonDisabled: true,
	showIccidAlert: false,
	showVerificationAlert: false,
	verifying: false
};

class ActivateSimModal extends React.PureComponent<ActivateSimModalProps, ActivateSimModalState> {
	static contextTypes: React.ValidationMap<ContextType> = contextTypesValidationMap;

	constructor(props: ActivateSimModalProps, context: ContextType) {
		super(props, context);
		this.state = {
			...initialState
		};
	}

	componentWillReceiveProps(newProps: ActivateSimModalProps) {
		if (this.state.verifying) {
			const { verificationResponse, verificationError } = newProps;

			const errorResponse =
				verificationError || (verificationResponse && verificationResponse.attributes.result !== "Success");

			if (errorResponse) {
				this.setState({
					showVerificationAlert: true,
					verifying: false
				});
			} else {
				this.onClose();
			}
		}
	}

	handleChange = (e: any) => {
		this.setState({
			error: !ERROR_RE.test(e.target.value),
			buttonDisabled: !DISABLED_RE.test(e.target.value),
			value: e.target.value
		});
	};

	handleActivate = () => {
		const activationInfo = this.props.simCardActivationInfo.find(info => info.iccid.endsWith(this.state.value));
		if (activationInfo) {
			if (activationInfo.iccidVerification) {
				this.props.simIccVerification({
					iccid: activationInfo.iccid,
					orderRefId: activationInfo.orderReference,
					successFlag: true
				});
				this.setState({
					verifying: true,
					showIccidAlert: false,
					showVerificationAlert: false
				});
			} else {
				this.onClose();
				window.open(activationInfo.tocTocUrl, "_blank");
			}
		} else {
			this.setState({
				showIccidAlert: true
			});
		}
	};

	onClose = () => {
		this.setState({
			...initialState
		});
		this.props.closeModal();
	};

	public getMainForm() {
		const inputClasses = classnames({
			inputNormal: true,
			error: this.state.error
		});
		return (
			<>
				<div className="verification-block">
					<label htmlFor="iccid-digits" className="label">
						<FormattedMessage { ...activateSimMessages.lastFiveDigits } />
					</label>
					<input
						className={inputClasses}
						id="iccid-digits"
						name="iccid-digits"
						value={this.state.value}
						onChange={this.handleChange}
						placeholder= {this.context.intl.formatMessage({ ...activateSimMessages.iccidDigitsPlaceholder })}
					/>
				</div>
				<div className="sim-block">
					<SimExampleView />
				</div>
			</>
		);
	}

	getContent() {
		const { showIccidAlert, showVerificationAlert } = this.state;
		return (
			<div className="container">
				{this.getMainForm()}
				{(showIccidAlert || showVerificationAlert) && (
					<div className="error-alert">
						<i className="fa fa-exclamation-triangle" />
						<FormattedMessage
							{...(showIccidAlert
								? activateSimMessages.invalidIccidMessage
								: activateSimMessages.verificationErrorMessage)}
						/>
					</div>
				)}
			</div>
		);
	}

	render() {
		return (
			<OcModal
				className="ActivateSim"
				showModal={this.props.show}
				largeModal={true}
				title={<FormattedMessage {...activateSimMessages.activateSimButton} />}
				onClose={this.onClose}
				onOk={this.handleActivate}
				okDisabled={this.state.buttonDisabled}
				okButtonLabel={<FormattedMessage {...activateSimMessages.activate} />}
			>
				{this.getContent()}
			</OcModal>
		);
	}
}

export default withCustomization(CommonCustomizationPoints.ACTIVATE_SIM_MODAL, ActivateSimModal);

export { ActivateSimModal, ActivateSimModalProps, ActivateSimModalState };
