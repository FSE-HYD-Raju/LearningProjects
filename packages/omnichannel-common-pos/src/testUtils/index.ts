import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
