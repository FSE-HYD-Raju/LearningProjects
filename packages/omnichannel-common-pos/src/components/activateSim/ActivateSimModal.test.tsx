import * as React from "react";
import { mountWithContext, shallowWithContext, TestUtils } from "../../testUtils";
import { ActivateSimModal, ActivateSimModalProps } from "./ActivateSimModal";
import { SimCardActivationInfo } from "./ActivateSimCard";

const { getModalContents } = TestUtils;

const simCardActivationInfo: SimCardActivationInfo[] = [
	{
		iccid: "8939883295001002529",
		orderReference: "CH231020181631",
		tocTocUrl:
			"http://toctoc.com?token=eyJhbGciOiJIUzI1NiJ9.eyJPcmRlcklkIjoiSGFyZEJfSG9tZURlbGl2ZXJ5XzE1MzgzODM2MzkifQ.q3DNQrYZpAVufkDU6CwCDhqnOe4f8n6v3-JEjwxq3zA",
		iccidVerification: true
	}
];

describe("ActivateSimModal", () => {
	let props: ActivateSimModalProps;
	beforeEach(() => {
		props = {
			show: true
		} as ActivateSimModalProps;
	});

	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(<ActivateSimModal {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
		mountWithContext(<ActivateSimModal {...props} />);
	});

	it("disables Activate button when input is not a 5-digits string", () => {
		const wrapper = mountWithContext(<ActivateSimModal {...props} />);
		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: "eeee" } });
		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		expect(btn.prop("disabled")).toEqual(true);
	});

	it("should disable Activate button when input is less then 5-digits", () => {
		const wrapper = mountWithContext(<ActivateSimModal {...props} />);
		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: "1234" } });
		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		expect(btn.prop("disabled")).toEqual(true);
	});

	it("should enable Activate button when input is 5-digits", () => {
		const wrapper = mountWithContext(<ActivateSimModal {...props} />);
		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: "12345" } });
		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		expect(btn.prop("disabled")).toEqual(false);
	});

	it("should show invalidIccidMessage in case of wrong inputted last digits", () => {
		props = {
			show: true,
			simCardActivationInfo
		} as ActivateSimModalProps;

		const wrapper = mountWithContext(<ActivateSimModal {...props} />);

		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: "12345" } });

		expect(wrapper.instance().state.showIccidAlert).toBeFalsy();

		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		btn.simulate("click");
		wrapper.update();

		expect(wrapper.instance().state.showIccidAlert).toBeTruthy();
		expect(wrapper.find(".ActivateSimModal-error-alert").find("span").props().children).toBe("Invalid last ICCID's digits");
	});

	it("should call closeModal after click on Cancel button", () => {
		const closeModal = jest.fn();
		props = ({
			show: true,
			closeModal
		} as any) as ActivateSimModalProps;

		const wrapper = mountWithContext(<ActivateSimModal {...props} />);

		const modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-cancel-button").hostNodes();
		btn.simulate("click");

		expect(closeModal).toBeCalled();
	});

	it("should call simIccVerification() after Activate click for right inputted iccid and sim_iccid verification", () => {
		const simIccVerification = jest.fn();
		props = ({
			show: true,
			simCardActivationInfo,
			simIccVerification
		} as any) as ActivateSimModalProps;

		const wrapper = mountWithContext(<ActivateSimModal {...props} />);

		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: "02529" } });

		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		btn.simulate("click");

		expect(wrapper.instance().state.verifying).toBeTruthy();
		expect(simIccVerification).toBeCalledWith({
			iccid: simCardActivationInfo[0].iccid,
			orderRefId: simCardActivationInfo[0].orderReference,
			successFlag: true
		});
	});

	it("should open TocToc url in new tab after Activate click for right inputted iccid and video verification", () => {
		const spy = jest.spyOn(window, "open");

		const closeModal = jest.fn();
		props = ({
			show: true,
			simCardActivationInfo: [
				{
					...simCardActivationInfo[0],
					iccidVerification: false
				}
			],
			closeModal
		} as any) as ActivateSimModalProps;

		const wrapper = mountWithContext(<ActivateSimModal {...props} />);

		let modalContents = getModalContents(wrapper);
		modalContents.find("input").simulate("change", { target: { value: "02529" } });

		modalContents = getModalContents(wrapper);
		const btn = modalContents.find("#oc-modal-action-button").hostNodes();
		btn.simulate("click");

		expect(closeModal).toBeCalled();
		expect(spy).toBeCalledWith(
			"http://toctoc.com?token=eyJhbGciOiJIUzI1NiJ9.eyJPcmRlcklkIjoiSGFyZEJfSG9tZURlbGl2ZXJ5XzE1MzgzODM2MzkifQ.q3DNQrYZpAVufkDU6CwCDhqnOe4f8n6v3-JEjwxq3zA",
			"_blank"
		);
	});

	it("should show error alert in case of failure simIccVerification response", () => {
		const wrapper = mountWithContext(<ActivateSimModal {...props} />);
		wrapper.setState({
			verifying: true
		});

		wrapper.setProps({
			verificationResponse: {
				attributes: {
					result: "Failure"
				}
			}
		});
		expect(wrapper.instance().state.verifying).toBeFalsy();
		expect(wrapper.instance().state.showVerificationAlert).toBeTruthy();
		expect(wrapper.find(".ActivateSimModal-error-alert").find("span").props().children).toBe("Error during ICCID validation");
	});

	it("should close modal in case of successful simIccVerification response", () => {
		const closeModal = jest.fn();
		props = ({
			show: true,
			closeModal
		} as any) as ActivateSimModalProps;

		const wrapper = mountWithContext(<ActivateSimModal {...props} />);
		wrapper.setState({
			verifying: true
		});

		wrapper.setProps({
			verificationResponse: {
				attributes: {
					result: "Success"
				}
			}
		});

		expect(closeModal).toBeCalled();
	});
});
