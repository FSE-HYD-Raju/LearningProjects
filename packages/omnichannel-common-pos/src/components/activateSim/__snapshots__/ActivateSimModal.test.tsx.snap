// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivateSimModal succeeds at shallow mount with minimum props 1`] = `
<OcModal
  className="ActivateSim"
  hideCloseButton={false}
  largeModal={true}
  okButtonLabel={
    <FormattedMessageWrapper
      defaultMessage="Proceed"
      description="Activate SIM"
      id="activate-sim-activate"
      wrapperType="wrapper"
    />
  }
  okDisabled={true}
  onClose={[Function]}
  onOk={[Function]}
  showModal={true}
  title={
    <FormattedMessageWrapper
      defaultMessage="Activate SIM"
      description="Activate SIM banner, activate button label"
      id="activate-sim-button-label"
      wrapperType="wrapper"
    />
  }
>
  <div
    className="ActivateSimModal-container"
  >
    <div
      className="ActivateSimModal-verification-block"
    >
      <label
        className="ActivateSimModal-label"
        htmlFor="iccid-digits"
      >
        <FormattedMessageWrapper
          defaultMessage="ICCID last 5 digits"
          description="Activate SIM modal, last five digits of ICCID message"
          id="activate-sim-last-five-digits-label"
          wrapperType="wrapper"
        />
      </label>
      <input
        className="ActivateSimModal-inputNormal"
        id="iccid-digits"
        name="iccid-digits"
        onChange={[Function]}
        placeholder="00025"
        value=""
      />
    </div>
    <div
      className="ActivateSimModal-sim-block"
    >
      <SimExampleView />
    </div>
  </div>
</OcModal>
`;
